### ISSUE TITLE
ModuleNotFoundError: No module named 'livy.models'; 'livy' is not a package

### ISSUE THREAD

**Issue by Bromania** [acroz/pylivy#63]
Hello, I'm trying to run some code for some personal project, but i got an error as said in the title. Here's the exact output :  

> Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/utils/autoreload.py", line 54, in wrapper
    fn(*args, **kwargs)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/core/management/commands/runserver.py", line 117, in inner_run
    self.check(display_num_errors=True)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/core/management/base.py", line 390, in check
    include_deployment_checks=include_deployment_checks,
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/core/management/base.py", line 377, in _run_checks
    return checks.run_checks(**kwargs)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/core/checks/registry.py", line 72, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/core/checks/urls.py", line 40, in check_url_namespaces_unique
    all_namespaces = _load_all_namespaces(resolver)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/core/checks/urls.py", line 57, in _load_all_namespaces
    url_patterns = getattr(resolver, 'url_patterns', [])
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/utils/functional.py", line 80, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/urls/resolvers.py", line 579, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/utils/functional.py", line 80, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/urls/resolvers.py", line 572, in urlconf_module
    return import_module(self.urlconf_name)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/root/Bastien/dev_frontend/site_test/site_test/urls.py", line 20, in <module>
    path('button/', include('button.urls')),
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/root/anaconda3/envs/tensorflow/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/root/Bastien/dev_frontend/site_test/button/urls.py", line 3, in <module>
    from . import views
  File "/root/Bastien/dev_frontend/site_test/button/views.py", line 3, in <module>
    import wordcount, livy
  File "/root/Bastien/dev_frontend/site_test/livy.py", line 3, in <module>
    from livy.models import Batches
ModuleNotFoundError: No module named 'livy.models'; 'livy' is not a package

I am currently using livy 0.6, I thought it was coming from this but since there is nothing about it anywhere I c an't be sure 100%. I am working with Python 3.6. (In this code i'm using Django to call livy)

Here is my code : 

```
Import json, pprint, requests, textwrap, logging
from typing import List
from livy.models import Batches


def send(inputText):
    host = 'http://localhost:8998'
    data = {"file":"/myapp/wordcount.py", "args":[inputText,"2"]}
    headers = {'Content-Type': 'application/json'}
    r = requests.post(host + '/batches', data=json.dumps(data), headers=headers)
    #session_url = host + r.headers['location']
    #getID = requests.get(host + '/batches' + 'id', data=json.dumps(data), headers=headers)
    #return getID

def sendAndGet(self) -> List[Batches]:
        """List all the active sessions in Livy."""
        data = self._client.get("/batches")
        return [Batches.from_json(item) for item in data["batches"]]
```

Since it's only a test, the last def. part is probably wrong but for now, it's only the livy.models import that I need.

I hope someone will be able to help me, thanks for your consideration.

By the way, the goal of this code is to get the ID of the livy batches i'm using for another purpose, if anyone has a better solution I am willing to listen.

Thank you very much.

**Comment by acroz** [comment]
Hi @Bromania - are you sure you've got the right library? The current released version of `pvlivy` does not have a `Batches` model.

Also, did you call your code file `livy.py`? The error suggests that there is a module which is not a package called `livy` in the import path (which includes the current working directory, and can  include the file you're running itself) which is not a package.

**Comment by Bromania** [comment]
Hi @acroz, thanks for your quick answer. First of all, I tried both `Sessions` and `Batches` but since none of them were working for me (same error) I sticked with `Batches` because I'm currently trying to work with it. Indeed, I'll most likely need to work with `Sessions` from now on.

Secondly, you are totally right, my code is named `livy.py`......... I feel so dumb about it, so thanks a lot I didn't pay attention to that. Thanks for the blog post, and thanks a lot for your answers I'll try it out and I'll tell you if I found another issue.
